### PR TITLE
fix: make `revalidateTags` no-op when list of tags is empty

### PR DESCRIPTION
**Pull request by eduardoboucas**
## Description

Makes the `revalidateTags` call a no-op when the array of tags is empty.

## Tests

Added integration test with scenario resulting in Next.js calling `CacheHandler.revalidateTag` with empty array of tags and ensuring purge API endpoint is not being used

## Relevant links (GitHub issues, etc.) or a picture of cute animal

https://linear.app/netlify/issue/FRB-1536/dont-call-purgecache-when-tags-is-empty-array

### CODE REVIEW

File: tests/integration/simple-app.test.ts [hunk 1]
```diff
@@ -22,6 +24,8 @@ import {
   startMockBlobStore,
 } from '../utils/helpers.js'
 import { nextVersionSatisfies } from '../utils/next-version-helpers.mjs'
+import { purgeCache } from '@netlify/functions'
+import { afterEach } from 'node:test'
```
Comment (serhalp) [inline tests/integration/simple-app.test.ts:28]: 🤔 shouldn't this be from `vitest` like all the others?

File: tests/integration/simple-app.test.ts [hunk 2]
```diff
@@ -210,6 +243,39 @@ test<FixtureTestContext>('cacheable route handler is cached on cdn (revalidate=f
...
+  // set the NETLIFY_PURGE_API_TOKEN to get pass token check and allow fetch call to be made
+  vi.stubEnv('NETLIFY_PURGE_API_TOKEN', 'mock')
+
+  const page1 = await invokeFunction(ctx, {
+    url: '/unstable_cache',
+  })
+  const data1 = load(page1.body)('pre').text()
+
+  // allow for cache entry to get stale
+  await new Promise((res) => setTimeout(res, 2000))
```
Comment (serhalp) [inline tests/integration/simple-app.test.ts:259]: is mocking and advancing the clock here problematic?

**Review by serhalp — APPROVED** [review]
LGTM

File: tests/integration/simple-app.test.ts (hunk 1 above)
Comment (pieh) [reply tests/integration/simple-app.test.ts:28]: oooh, yes

File: tests/integration/simple-app.test.ts (hunk 2 above)
Comment (pieh) [reply tests/integration/simple-app.test.ts:259]: we use above in plenty of other places that check behaviors related to stale cache items so I just the same method here

I'm not sure if attempting to mock time would work

File: tests/integration/simple-app.test.ts (hunk 1 above)
Comment (pieh) [reply tests/integration/simple-app.test.ts]: https://github.com/opennextjs/opennextjs-netlify/pull/2727/commits/4a02e345b378a22fec355ced93a1e9ffa9886a87

**Review by pieh — APPROVED** [review]

**Pull request merged**